compiler/rustc_codegen_ssa/src/codegen_attrs.rs

```diff
@@ -463,6 +466,59 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, did: LocalDefId) -> CodegenFnAttrs {
                     None
                 };
             }
+            sym::patchable_function_entry => {
+                codegen_fn_attrs.patchable_function_entry = attr.meta_item_list().and_then(|l| {
+                    let mut prefix = None;
+                    let mut entry = None;
+                    for item in l {
+                        let Some(meta_item) = item.meta_item() else {
+                            tcx.dcx().span_err(item.span(), "Expected name value pair.");
+                            continue;
+                        };
+
+                        let Some(name_value_lit) = meta_item.name_value_literal() else {
+                            tcx.dcx().span_err(item.span(), "Expected name value pair.");
+                            continue;
+                        };
+
+                        let attrib_to_write = match meta_item.name_or_empty() {
+                            sym::prefix_nops => &mut prefix,
+                            sym::entry_nops => &mut entry,
+                            _ => {
+                                tcx.dcx().span_err(
+                                    item.span(),
+                                    format!(
+                                        "Unexpected parameter name. Allowed names: {}, {}",
+                                        sym::prefix_nops,
+                                        sym::entry_nops
+                                    ),
+                                );
+                                continue;
+                            }
+                        };
+
+                        let rustc_ast::LitKind::Int(Pu128(val @ 0..=255), _) = name_value_lit.kind
+                        else {
+                            tcx.dcx().span_err(
+                                name_value_lit.span,
+                                "Expected integer value between 0 and 255.",
+                            );
```

Comment:
```suggestion
                            tcx.dcx().struct_span_err(
                                name_value_lit.span,
                                "integer value out of range",
                            ).span_label(
                                name_value_lit.span,
                                "value must be between `0` and `255`",
                            ).emit();
```

Reply:
This error message would also be triggered when passing in a string, so the suggested wording might be a bit weird in that case. Do you think it makes sense to make the check for the string seperate again or is this just a minor issue?

Reply:
I think it is worth it to be more accurate whenever possible. you can make this be

```suggestion
                            let mut err = tcx.dcx().struct_span_err(
                                name_value_lit.span,
                                "invalid literal value",
                            );
                            if is_numeric {
                                err.span_label(
                                    name_value_lit.span,
                                    "value must be between `0` and `255`"
                                );
                            } else if is_str {
                                err.span_label(
                                    name_value_lit.span,
                                    "some appropriate message"
                                );
                            }
                            // Are there other cases to handle? can we hit this error without a numeric or str literal?
                            err.emit();
```